### PR TITLE
Create (S)CSS-flavored Flame implementation package

### PR DESCRIPTION
**Pull request by xdrdak**
## Description

Full port of the SASS version of flame that was developed by Chronogolf.

This is a simply lift and shift and groundworks preparation for the next stuff.

Do note, we DO NOT plan to deprecate `flame` or to stop support. The intent of flame-css is to provide the basic building blocks for creating Lightspeed experiences, as well as providing a bunch of utility functions (which will get improved) for frameworks/projects that are incapable of supporting React.

Although we are bundling the sass source code, for the most part, you should only be using the generated css file.

## How to test?

- Checkout branch, go to the flame-css package and run `yarn dev`
- Open [Storybook](http://localhost:6006)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/lightspeed/flame/blob/master/.github/CONTRIBUTING.md) guide
- [x] I have prepared [CHANGELOGs](https://github.com/lightspeed/flame/blob/master/.github/CONTRIBUTING.md#git-workflow) for release
- [x] I have tested my changes on [supported browsers](https://browserl.ist/?q=%3E0.25%25%2C+not+op_mini+all%2C+not+ie+%3C%3D+11)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by xdrdak** [comment]
fyi if you are going to be using this package, chances are it will be subject to change.

Nothing zany like complete classname overhauls, but most likely namespacing/imports.

File: packages/flame-css/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+Refer to the [CONTRIBUTING guide](https://github.com/lightspeed/flame/blob/master/.github/CONTRIBUTING.md) for more info.
+
+## [Unreleased]
+
+### Added
+
+- Port over sassy version of flame
```
Comment (glambert) [inline packages/flame-css/CHANGELOG.md:14]:
```suggestion
- Create base Flame CSS implementation package ([#109](https://github.com/lightspeed/flame/pull/109))
```

File: packages/flame-css/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Flame css
+
+`flame-css` is an alternative offering to `flame`.
+
+It offers a sass codebase and an easy to consume css file.
```
Comment (glambert) [inline packages/flame-css/README.md:5]: I would avoid mentioning Sass in the intro. Perhaps later down as an "Unsupported" feature that we might drastically change or remove at some point.

File: packages/flame-css/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Flame css
```
Comment (glambert) [inline packages/flame-css/README.md:1]:
```suggestion
# Flame CSS
```

File: packages/flame-css/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+`flame-css` is an alternative offering to `flame`.
+
+It offers a sass codebase and an easy to consume css file.
+
+@TODO:
+
+- [ ] Hook flame-tokens/flame-foundations as primary variables
+- [ ] Integrate tailwind to output utility classes
```
Comment (glambert) [inline packages/flame-css/README.md:10]: Let's rephrase this to avoid implementation details. Let's instead mention that we'd like to include a guide on how to integrate [utility-first CSS](https://css-tricks.com/in-defense-of-utility-first-css/) on top of Flame CSS.

File: packages/flame-css/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- [x] Storybook setup
+- [x] Write all stories
+- [ ] Snapshot testing
+- [ ] Publishing
```
Comment (glambert) [inline packages/flame-css/README.md:14]: Publishing will be done as soon as we merge/`yarn release` right?

File: packages/flame-css/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+## Caveat
+
+Although we are bundling the sass source code, for the most part, you should only be using the generated css file.
```
Comment (glambert) [inline packages/flame-css/README.md:18]:
```suggestion
Although we are bundling the Sass source code,  you should only be using the generated CSS file in the majority of cases.
```

Ah, here's what I mentioned in the README intro. I would move this at the top so that it's clear from the start.

File: packages/flame-css/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# Flame css
+
+`flame-css` is an alternative offering to `flame`.
```
Comment (glambert) [inline packages/flame-css/README.md:3]:
```suggestion
A pure CSS alternative offering to [Flame React](https://github.com/lightspeed/flame#readme).
```

File: packages/flame-css/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+However, `flame-css` provides all the necessary basic components and utility functions to build out the complex
+functionalities in whatever framework you choose.
+
+## Installing
+
+If you are leveraging some form of bundler (webpack, parcel...), you may wish to import the compiled css file
+into the root of your app.
+
+Please ensure you have the appropriate loaders before importing a `.css` file
```
Comment (glambert) [inline packages/flame-css/README.md:31]: Link to webpack loaders?

File: packages/flame-css/README.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+`flame-css` is not meant to be a complete replacement to `flame`, as it lacks many more complex functionalities
+that just are not possible without some front-end framework.
+
+However, `flame-css` provides all the necessary basic components and utility functions to build out the complex
+functionalities in whatever framework you choose.
+
+## Installing
+
+If you are leveraging some form of bundler (webpack, parcel...), you may wish to import the compiled css file
```
Comment (glambert) [inline packages/flame-css/README.md:28]: I'd make it clear this is the recommended way, not "if you are leveraging...".

File: packages/flame-css/README.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+Although we are bundling the sass source code, for the most part, you should only be using the generated css file.
+
+`flame-css` is not meant to be a complete replacement to `flame`, as it lacks many more complex functionalities
+that just are not possible without some front-end framework.
+
+However, `flame-css` provides all the necessary basic components and utility functions to build out the complex
+functionalities in whatever framework you choose.
+
+## Installing
```
Comment (glambert) [inline packages/flame-css/README.md:26]: Let's move this above the `caveat` section, and add in a step for [font setup](https://github.com/lightspeed/flame#link-fonts) (I'd copy/paste in case Flame React mainline changes font setup along the way).

File: packages/flame-css/package.json [hunk 11]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@lightspeed/flame-css",
+  "version": "0.0.0",
+  "private": true,
```
Comment (glambert) [inline packages/flame-css/package.json:4]: FYI I'd start with `0.1.0` when we `yarn release`. See https://github.com/lightspeed/flame/blob/next/RELEASE.md#note-on-new-packages.

File: packages/flame-css/package.json [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  "description": "Base lightspeed components, but in css!",
```
Comment (glambert) [inline packages/flame-css/package.json:5]:
```suggestion
  "description": "Base Lightspeed helpers and components CSS",
```

File: packages/flame-css/src/mixins/media-queries.scss [hunk 13]
```diff
@@ -0,0 +1,52 @@
+/*
+  Media Queries
+  ======
+  A mobile first approach for responsive UI.
+*/
+
+$fl-tablet-portrait-width: 600px;
+$fl-tablet-landscape-width: 900px;
+$fl-tablet-desktop-width: 1200px;
+$fl-tablet-big-desktop-width: 1800px;
```
Comment (glambert) [inline packages/flame-css/src/mixins/media-queries.scss:10]: Those should follow breakpoints we have in tokens: https://github.com/lightspeed/flame/blob/master/packages/flame-tokens/src/theme-ui/common.ts#L4

File: packages/flame-css/README.md (hunk 5 above)
Comment (xdrdak) [reply packages/flame-css/README.md:14]: gonna remove that particular TODO as it was only pertinent for my personal tracking.

File: packages/flame-css/src/mixins/media-queries.scss (hunk 13 above)
Comment (xdrdak) [reply packages/flame-css/src/mixins/media-queries.scss:10]: removing this entire file as this is opening a can of worms that I do not want to tackle in this PR
